Issue building the main branch int GitHub Actions runner (macOS 15 Intel)

Some time ago I promised to setup stuff such that we can build/test MPICH on GitHub actions. Since then, I've been working in spare time to set things up. The idea is not to replace current MPICH testing infrastructure, but rather use it as a quick-turnover barrier for trivial build/test issues that may go unnoticed. My work so far is here: https://github.com/dalcinl/ci-mpich. 
This work is almost ready, the only thing to decide about are:
a) whether to run the full MPICH test suite, due to the time it may take.
b) whether to add mpi4py tests to the workflow, which should add a few minutes (no more than 5).
For reference Open MPI has been running mpi4py test in their GitHub Actions for quite some time.

However, before posting that work for your consideration, I'm in need of your help to deal with a build issue.
I'm facing issues building on `macOS 15` and `macOS 26`, but only for the case of `x86_64`, here you have the full logs:
https://github.com/dalcinl/ci-mpich/actions/runs/27696157870/job/81919807149

The failure a linking step for the Fortran library:
```console
ld: warning: -dylib_file is deprecated. Use -F or -L to control where indirect dylibs are found
ld: warning: -dylib_file is deprecated. Use -F or -L to control where indirect dylibs are found
ld: fixup error (kind=x86_64_rip) at '_MPL_check_arch_features'+0x16 from mpl_arch_x86.o, target '___cpu_model' does not have address
final section layout:
    __TEXT               addr=0x000000000, size=0x000047000, fileOffset=0x00000000, fileSize=0x00047000
        __text           addr=0x000000990, size=0x000031a95, fileOffset=0x00000990
        __stubs          addr=0x000032425, size=0x000001908, fileOffset=0x00032425
        __stub_helper    addr=0x000033d2d, size=0x0000029c8, fileOffset=0x00033d2d
        __cstring        addr=0x000036700, size=0x00000133e, fileOffset=0x00036700
        __const          addr=0x000037a40, size=0x000000070, fileOffset=0x00037a40
        __eh_frame       addr=0x000037ab0, size=0x00000f068, fileOffset=0x00037ab0
    __DATA_CONST         addr=0x000047000, size=0x000001000, fileOffset=0x00047000, fileSize=0x00001000
        __got            addr=0x000047000, size=0x0000000e8, fileOffset=0x00047000
    __DATA               addr=0x000048000, size=0x00000a000, fileOffset=0x00048000, fileSize=0x00003000
        __la_symbol_ptr  addr=0x000048000, size=0x000002160, fileOffset=0x00048000
        __data           addr=0x00004a160, size=0x000000518, fileOffset=0x0004a160
        __common         addr=0x00004a680, size=0x000000132, fileOffset=0x00000000
        __bss            addr=0x00004a7c0, size=0x0000073d0, fileOffset=0x00000000
    __LINKEDIT           addr=0x000052000, size=0x00003b000, fileOffset=0x0004b000, fileSize=0x0003a258
collect2: error: ld returned 1 exit status
make[2]: *** [lib/libmpifort.la] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

As the failure seems related to `MPL_check_arch_features()`, I'm wondering whether #7699 by @raffenet may be related.

Note however that building the latest release tarball is fine in either Intel and Apple silicon.
For example, see https://github.com/mpi4py/mpi-publish/actions/runs/27575133820/job/81521517781#step:9:317

--- a/src/mpl/src/arch/mpl_arch_x86.c
+++ b/src/mpl/src/arch/mpl_arch_x86.c
@@ -6,6 +6,10 @@
 #include "mpl.h"
 #include "mpl_base.h"
 
+#ifdef __APPLE__
+#undef MPL_HAVE_BUILTIN_CPU_SUPPORTS
+#endif
+
 uint64_t mpl_arch_features = MPL_ARCH_FLAG__UNKNOWN;
 
 #ifndef MPL_HAVE_BUILTIN_CPU_SUPPORTS

